README.md

```diff
@@ -385,21 +387,46 @@ There are two built-in casting operators in **Jakt**.
+- [ ] Casts to the same type are infallible and pointless, so might be forbidden in the future.
+- [ ] If both types are primitive, a safe conversion is done.
+    - [ ] Integer casts will fail if the value is out of range. This means that promotion casts like i32 -> i64 are infallible.
+    - [ ] Float -> Integer casts truncate the decimal point, if the float is a normal numeric value. If the float is any NaN or either Infinity, the cast fails.
+    - [ ] Integer -> Float casts resolve to the closest value to the integer representable by the floating-point type. If the integer value is not exactly representable, the cast fails.
+    - [ ] Any primitive -> bool will create `true` for any value except 0, which is `false`.
+    - [ ] bool -> any primitive will do `false -> 0` and `true -> 1`, even for floats.
+- If the types are two different pointer types of the same underlying type `T`, the cast is generally allowed and infallible.
+    - [ ] A cast to `T` from any pointer type, if `T` is reference counted, will increment the reference count as expected; that's the preferred way of creating a strong reference from a weak or temporary reference.
+    - [ ] A cast to `T` from any pointer type, if `T` is *not* reference counted, will copy the value.
+    - [ ] Casts from `&T` to `&mut T` is forbidden. The other way around is allowed.
+    - [ ] A cast from and to `raw T` is unsafe but infallible.
+    - [ ] A cast to `&T` or `&mut T` is allowed for orthagonality, but has the same effect as directly taking the reference with the `&` or `&mut` operators, respectively.
+    - [ ] Casts from `&T` or `&mut T` are allowed for orthagonality and follow the same rules as dereferencing.
+    - [ ] Casts between infallible pointer types can happen alongside casts within the contained type, if the pointer type doesn't change or is cast strictly upwards on the pointer type hierarchy. This hierarchy goes (from bottom to top) `&mut T` -> `&T` -> `T`.
+- If the types are part of the same type hierarchy (i.e. one is a child type of another), casts are possible according to these rules. Note that this also goes if the types are the same pointer type and the contained type is part of the same type hierarchy (which applies recursively), so `&mut A` and `&mut B` are part of the same type hierarchy if `A` and `B` are.
+    - [ ] A child can be cast to its parent infallibly.
+    - [ ] A parent can be cast to a child, but this will check the type at runtime and fail if the object was not of the child type or one of its subtypes.
+- [ ] If the types are incompatible, a user-defined cast is attempted to be used. The details here are not decided yet, but it will involve implementing some generic trait.
+- If nothing works, the cast will not compile.
+
+Some examples of possible and not possible casts (the left side shows the data type of a `value`, not the `value` itself):
```

Comment:
but nothing indicates what's possible and what's not?

Reply:
The comments do?

Reply:
All of them read as "this is possible" to me,
> i32 as! &i32 // equivalent to (&value)

Reply:
Ah right, what I mean by that is that "semantics of `&value` apply", i.e. it does not compile if `&value` does not compile.